### DISCUSSION TITLE
Immich “Spaces” - support multiple sanboxed fully-featured “libraries”

### DISCUSSION THREAD

**Discussion by mist2t** [immich-app/immich#12444] [General]
1. Some of us are doing photography professionaly

2. Others are doing it as a hobby

3. Pretty much all of us have ALSO their family pictures (smartphone snapshots, old scans, pictures from old point-and-shoot cameras etc) ... on top of everything else.

I couldn't find any software designed to properly "segregate" functionality based on sandboxed "spaces" defined by user. 

Let me explain ...

1. Ability to define "spaces" and easy switch between those
2. A proper sandboxed space: i don't want to see Faces from my professional photography together with those from my family archive.

I don't want to see my "Professional" photographs on the map together with my family photos, similar for faces, tags, albums, timeline ... etc You get the idea !

I've been looking for a very long time for a solution that will accommodate this need:

A fully featured image management software (with cool stuff like: places, people, AI etc.) BUT "segregated" and sandboxed on particular needs (family, clients photography etc.)

Pretty much all the software out there uses a single "library" and will pull data from this single “library” thus combining everything together in a huge messy "soup" (clients faces and their gazillions relatives from weddings .... with my personal family faces for example)

Having the ability to "categorize" / segregate on a deeper level and define properly sandboxed "spaces" .... will be HUGE for those users that are a little bit more invested in photography .

Until then, the only option is pretty much to stop using some functionalities (faces becomes unbelievable bloated when you have your clients, hobby photography and family together) or just use the app for a particular scenario (only for family photos archive for example)

Let me know what you think ! Thx :)

**Comment by bo0tzz** [comment]
You can accomplish this pretty easily by having separate users.

**Reply by mist2t** [reply]
Juggling in-and-out between multiple user accounts it's not a great experience. 

I'm also not sure how this works with external libraries ...

Anyway i think, long-term, "professional" scenarios should also be taken into consideration.

A lot of us go beyond smartphone pictures 🙂

**Comment by radh21301** [comment]
Maybe the same post? Either way, linking it here for reference.

https://www.reddit.com/r/immich/s/pZp7ejOlWf

**Comment by mist2t** [comment]
Wow ... somebody was so intrigued by this idea and closed it instantly 👍 🙂

**Reply by radh21301** [reply]
I suppose multi accounts/users should get what you want right?

If you are looking for easily switching between accounts, I found another feature request. You could probably add on to the comments there.

https://github.com/immich-app/immich/discussions/2919
